internal/server/auth/middleware.go

```diff
@@ -137,6 +138,45 @@ func UnaryInterceptor(logger *zap.Logger, authenticator Authenticator, o ...cont
 	}
 }
 
+// EmailMatchingInterceptor is a grpc.UnaryServerInterceptor only used in the case where the user is using OIDC
+// and wants to whitelist a group of users issuing operations against the Flipt server.
+func EmailMatchingInterceptor(logger *zap.Logger, emailMatches []string) grpc.UnaryServerInterceptor {
+	return func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
+		auth := GetAuthenticationFrom(ctx)
+
+		if auth == nil {
+			return handler(ctx, req)
+		}
+
+		email, ok := auth.Metadata["io.flipt.auth.oidc.email"]
+		if !ok {
```

Comment:
are emails always returned from oidc providers? iirc i thought the user had to request it, or maybe we have it defaulted? I don't think we want to skip this check if there isn't an email, we likely want to return unauthorized if the operator has emails in the allow list but did not ask for the `email` scope

Reply:
+1 if the email list has been specified I would err on the side of closed rather than open

Reply:
@markphelps Yeah I was thinking about this as well. I will switch the logic up.